scripts/benchmarks/csv-json-converter.ts

```diff
@@ -0,0 +1,132 @@
+/**
+* @CsvToJsonConverter
+*  This is a utility script which converts change request CSVs
+* to json.
+*/
+import _ from 'lodash';
+import { parse } from 'csv-parse/sync';
+
+import {
+    BENCHMARKS_COLUMN_NAMES,
+    BENCHMARKS_ORDER,
+    BOOLEAN_CSV_FIELDS,
+    SUBMISSION_METHOD_MAP,
+} from '../constants';
+import { InvalidValueError } from '../errors';
+
+type Benchmark = {
+    measureId?: string,
+    benchmarkYear?: number,
+    performanceYear?: number,
+    submissionMethod?: string,
+    isToppedOut?: boolean,
+    isHighPriority?: boolean,
+    isInverse?: boolean,
+    metricType?: string,
+    isToppedOutByProgram?: boolean,
+    percentiles?: object,
+}
+
+export function convertCsvToJson(csv: any, performanceYear: number) {
+    const parsedCsv = prepareCsv(csv);
+
+    let mappedCsv = parsedCsv.map((row: any) => {
+        const measure: Benchmark = {};
+        //maps the csv column values to the matching measures-data fields.
+        _.each(BENCHMARKS_COLUMN_NAMES, (columnName, measureKeyName) => {
+            if (row[measureKeyName]) {
+                measure[columnName] = mapInput(measureKeyName, row);
+            }
+        });
+        if (row['percentile10']) {
+            measure.percentiles = {
+                "1": row['percentile1'],
+                "10": row['percentile10'],
+                "20": row['percentile20'],
+                "30": row['percentile30'],
+                "40": row['percentile40'],
+                "50": row['percentile50'],
+                "60": row['percentile60'],
+                "70": row['percentile70'],
+                "80": row['percentile80'],
+                "90": row['percentile90'],
+                "99": row['percentile99'],
+            }
+        }
+        
+        //populate with False if it wasn't found in the csv.
+        measure.isToppedOutByProgram = measure.isToppedOutByProgram || false;
+        
+        measure.performanceYear = performanceYear;
+        measure.benchmarkYear = performanceYear - 2;
+        
+        return orderFields(measure);
+    });
+
+    //remove any empty rows
+    _.remove(mappedCsv, _.isEmpty);
+
+    return mappedCsv;
+}
+
+function orderFields(measure: any) {
+    //reorder the fields to stay consistent.
+    const orderedValues = Object.assign({}, BENCHMARKS_ORDER, measure)
+    //remove undefined fields.
+    return Object.entries(orderedValues)
+        .filter(([key, value]) => value !== undefined)
```

Comment:
This block could be cleaned up with 
`return _.filter(orderedValues, (value) => { value !== undefined }));`